### ISSUE TITLE
comma in data value is not escaped

### ISSUE THREAD

**Issue by azrdev** [renatopp/liac-arff#15]
, in any data field should lead to this field being quoted with ', but doesn't

**Comment by renatopp** [comment]
This is an issue related to python `csv` reader, I need some help to find out how to enable quoting of multiple characters `"` and `'`

**Comment by midnightradio** [comment]
Basically, values are delimited by comma no matter the type of each values are (and quoted or doubleqouted is only the case for string typed data) and comma inside a quoted string or nomial data should be escaped in ARFF file. So in this case, I think it needs preprocessing escapes comma inside a quoted string before it read by the ARFF decoder.

Since ARFF does not specify any character to be used for escaping, just following popular case with '\' for allowing the preprocessors to deal with some escaped characters would be enough. In that case, changing the decoder is necessary with giving csv.reader a parameter 'escapechar'.
```
In [4]: for row in csv.reader(['one,two,three\,four'], escapechar='\\'):
    print row
   ...:     
['one', 'two', 'three,four']
```

**Comment by dalupus** [comment]
I ran into this today I think.   

```
@relation reviewSpamData-string-noSentiment
@attribute text string
@attribute spam {N,Y}
@data
'Doctor Aboolian is a board-certified, plastic surgeon.',Y
```

This results in an error as it tries to break that string into 2.

**Comment by dalupus** [comment]
For some reason it is escaping the opening and closing ' which is confusing csv reader
\\'Doctor Aboolian is a board-certified, plastic surgeon.\\',Y

That is what ends up as the string it tries to read.  since the open and close ' is escaped it doesn't treat it as a singular field.

**Comment by dalupus** [comment]
Doing this on line 334 fixed it for me. In my case.  csv.reader defaults to using double quote as the quote char and none as the escapechar.

`values = next(csv.reader([row.strip(' ')],quotechar="'",escapechar="\\"))`

**Comment by dalupus** [comment]
The [official](http://weka.wikispaces.com/ARFF+%28developer+version%29) spec says that double or single quotes are allowed (I think it used to only allow single quotes), which csv.reader does not allow this.  Maybe we need to make this a parameter which can be passed in?

I believe that single quotes is the norm for arff files though, atleast in 3.6.

**Comment by joaquinvanschoren** [comment]
Is there any workaround for this issue? This really doesn't make any sense.
Even if I clearly state that an attribute is a string, and pass it as a string, it is converted to multiple attributes...

```
my_arff = {"attributes": [["attr","STRING"]], "data": [["a,b,c"],["a,b,c"]], "relation": "bla"}
with open("text.arff", "w") as fp:
    arff.dump(my_arff, fp)
```
Yields:
```
@RELATION bla
@ATTRIBUTE attr STRING
@DATA
a,b,c
a,b,c
```

**Comment by mfeurer** [comment]
This indeed seems to be problematic. What kind of behavior would you expect? Add single quotes when dumping a string? Checking whether parsing the string would lead to the same result?

**Comment by joaquinvanschoren** [comment]
Quoting strings when dumping string attributes would indeed help.

I also noticed that adding a single space at the end gives you the correct behaviour, as in:
```
my_arff = {"attributes": [["attr","STRING"]], "data": [["a,b,c "],["a,b,c "]], "relation": "bla"}
```

**Comment by mfeurer** [comment]
@joaquinvanschoren I fixed your issue in commit 0248228e6de857e2728a845e9536ec18bcbf847e. Could you please check and confirm? Also, I'll try to make sure the other issue goes away as there are already two PRs on that issue.

**Comment by joaquinvanschoren** [comment]
Looks good!

**Comment by mfeurer** [comment]
Fixed in #63.
